### ISSUE TITLE
[BACKPORT] Add OutBound Type to aks_config_v2

### ISSUE THREAD

**Issue by matttrach** [rancher/terraform-provider-rancher2#1443]
Issue tracking backport of issue #1442 PR #1439 which addresses https://github.com/rancher/aks-operator/issues/721

**Comment by markusewalker** [comment]
Reopening this issue. When trying to run `terraform apply`, I get the following:

```
│ Error: Unsupported argument
│ 
│   on main.tf line 29, in resource "rancher2_cluster" "cluster":
│   29:     outbound_type              = var.outbound_type
│ 
│ An argument named "outbound_type" is not expected here. Did you mean "outboung_type"?
```
Looking at the code, the typo `outboung_type` is there. FWIW, this change works, but the typo needs to be fixed to have `outbound_type` before QA will close this issue.

**Comment by jiaqiluo** [comment]
The fix can be validated on https://github.com/rancher/terraform-provider-rancher2/releases/tag/v5.2.0-rc2

**Comment by markusewalker** [comment]
Validated this is fixed on `v5.2.0-rc2`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.9-head` https://github.com/rancher/rancher/commit/9a7e264b25b9a2e5a68d41af61386aa23b953175
- Rancher2 version: `v5.2.0-rc2`

**TEST RESULTS**
|#|Scenario|Result|
|---|---|---|
|1|Provision AKS cluster with new parameter `outbound_type`|:white_check_mark:|
